core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -1047,6 +1047,8 @@ sealed trait WitnessScriptPubKey extends ScriptPubKey {
   def witnessVersion = WitnessVersion(asm.head)
 }
 
+sealed trait BIP143VulnerableScriptPubKey extends WitnessScriptPubKey
```

Comment:
Probably worth a doc here explaining the vulnerability

Reply:
Also is there any reason to introduce this trait rather than just using `WitnessScriptPubKeyV0`?